src/Cassandra.IntegrationTests/OpenTelemetry/OpenTelemetryTests.cs

```diff
@@ -0,0 +1,150 @@
+﻿using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using Cassandra.OpenTelemetry;
+using Cassandra.Tests;
+using NUnit.Framework;
+using OpenTelemetry;
+using OpenTelemetry.Trace;
+
+namespace Cassandra.IntegrationTests.OpenTelemetry
+{
+    [Category(TestCategory.Short)]
+    public class OpenTelemetryTests : SharedClusterTest
```

Comment:
Can we have a test that verifies that the activity context is correctly propagated? I don't think there is one at the moment that verifies this unless I missed it. If I'm not mistaken one of the use cases for this is to properly correlate these traces with (for example) an API call so the context needs to be properly propagated correct? Ideally we would have a test for statements, one for the mapper and one for LINQ. And each test should use both sync and async variants of the execute methods.

We should also test that the context propagation works correctly with auto paging, although I'm not 100% clear what behavior we want in this case. I'm assuming the context we want is the one that is active the moment where the enumerator pages the next page so the current changes might already do that but we need a test for it.

Reply:
I agree. The context propagation tests can be straightforward. I will include a method that will be called after the database requests and check if the traceId is the same and also that the database span id is the parent of the method span id.

Reply:
Working on the tests currently 👍

Reply:
Let me know when I can take a look at the tests 👍

Reply:
Currently I have the following tests:

- When the Keyspace is available check if DbOperation and DbName attributes are included;
- When the Keyspace is not available available check if DbName attribute is not included;
- With default builder options, the statement is not included as an attribute ;
- With the option `IncludeDatabaseStatement` the statement is included as an attribute;
- A test with `Execute` and `Execute` async methods in which I verify if the trace is created and also if the node span is child of session span;
- A test with `Mapper` and `Mapper` async methods in which I verify if the trace is created and also if the node span is child of session span;

Currently missing is the LINQ test and the context propagation that you mentioned above.
I also wanna make a test with a retry to see if multiple spans are generated that are child of the same session span.

I will try to finish in the next few days so, if you have time available, you can start checking the current tests :)

Reply:
@joao-r-reis I finished the tests. Since my previous comment I included
- A test to validate the activity when executing using LINQ
- A test to validate if the same TraceId is used when invoking a different method with a call to Cassandra in the mix
- A test to validate if, when a host is down and a retry happens, if it is produced more than one node activity for the same session activity. During this test, I realized that if multiple retries happens on the same host, the node level activity would be reused instead of being created a new one, so I fixed it in this commit https://github.com/datastax/csharp-driver/pull/608/commits/13f3eec6001b1a5cc74f5cf155445d93037d0b6a
- A test for pagination